core/src/main/java/org/apache/iceberg/PartitionsTable.java

```diff
@@ -93,16 +96,77 @@ private static StaticDataTask.Row convertPartition(Partition partition) {
     return StaticDataTask.Row.of(partition.key, partition.recordCount, partition.fileCount, partition.specId);
   }
 
-  private static Iterable<Partition> partitions(StaticTableScan scan) {
+  private static Iterable<Partition> partitions(Table table, StaticTableScan scan) {
     CloseableIterable<FileScanTask> tasks = planFiles(scan);
+    Types.StructType normalizedPartitionType = Partitioning.partitionType(table);
+    PartitionMap partitions = new PartitionMap();
+
+    LoadingCache<Integer, Integer[]> originalPartitionFieldPositionsBySpec = Caffeine.newBuilder().build(specId ->
```

Comment:
I think the number of partition specs should be very small so I would probably just use a 
Maps.newHashMap()

Then .computeIfAbsent

Not that I think there is somethign wrong with Caffine, just seems a bit heavy weight to me in this use case